lib/nimble_parsec.ex

```diff
@@ -158,14 +158,8 @@ defmodule NimbleParsec do
           @compile {:inline, inline}
         end
 
-        if combinator_kind == :def do
-          for {name, args, guards, body} <- defs do
-            def unquote(name)(unquote_splicing(args)) when unquote(guards), do: unquote(body)
-          end
-        else
-          for {name, args, guards, body} <- defs do
-            defp unquote(name)(unquote_splicing(args)) when unquote(guards), do: unquote(body)
-          end
+        for {name, args, guards, body} <- defs do
+          def unquote(name)(unquote_splicing(args)) when unquote(guards), do: unquote(body)
```

Comment:
It is a small detail but one is `def`, the other is `defp`. :)

Reply:
@josevalim  I guess I shall apologize - need new glasses - but actually the tests still passed, shall I change the PR to just add a test for that case (and leave the code unchanged of course)?

Reply:
@josevalim  maybe I am wrong, but I guess there is something wrong here (to not abuse of the proverbial smell in a scandinavian country as mentioned in a play by Shakespear)

I just put an `IO.inspect` at the end of the `for`:

```
[parse__0: 6, parse__0: 6, parse__1: 6]
[parsep__0: 6, parsep__0: 6, parsep__1: 6]
[combinator__0: 6, combinator__0: 6, combinator__1: 6]
[combinatorp__0: 6, combinatorp__0: 6, combinatorp__1: 6
```
 is the AST that gets created here

While the b

Reply:
A PR that improves the coverage would certaily be welcome!

Reply:
I will try...